packages/docusaurus-theme-common/src/contexts/doc.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useMemo, type ReactNode, useContext} from 'react';
+import {ReactContextError} from '../utils/reactUtils';
+import type {PropDocContent} from '@docusaurus/plugin-content-docs';
+
+export type DocContextValue = {MDXComponent: () => JSX.Element} & Pick<
```

Comment:
not sure how to name this context/hook/file

`useDoc` doesn't look too bad and is relatively simple

Reply:
I think it's fine. Need more JSDoc on how you are supposed to use this